src/java.desktop/share/classes/javax/swing/filechooser/FileSystemView.java

```diff
@@ -262,32 +262,41 @@ public Icon getSystemIcon(File f) {
     * <p>
     * Example: <pre>
     *     FileSystemView fsv = FileSystemView.getFileSystemView();
-    *     Icon icon = fsv.getSystemIcon(new File("application.exe"), 64);
+    *     Icon icon = fsv.getSystemIcon(new File("application.exe"), 64, 64);
     *     JLabel label = new JLabel(icon);
     * </pre>
     *
-    * @implSpec The default implementation is platform specific
-    * and while we will do our best to find an icon of a specific size the exact
-    * match can not be guaranteed. Wherever supported by the underlying
-    * implementation, the icon returned is a multi-resolution icon image,
+    * @implSpec The available icons may be platform specific and so the
+    * available sizes determined by the platform. Therefore an exact match
+    * for the requested size may not be possible.
+    *
+    * The icon returned may be a multi-resolution icon image,
     * which allows better support for High DPI environments
     * with different scaling factors.
     *
     * @param f a {@code File} object for which the icon will be retrieved
-    * @param size width and height of the icon in user coordinate system.
-    * This API only supports square icons with the edge length
-    * equals or greater than 1. Maximum size allowed is defined by the
-    * underlying implementation.
+    * @param width width of the icon in user coordinate system.
+    * @param height height of the icon in user coordinate system.
     * @return an icon as it would be displayed by a native file chooser
-    * or null if invalid parameters are passed such as pointer to a
-    * non-existing file or a size outside of supported range.
+    * or null if invalid parameters are passed such as reference to a
+    * non-existent file.
+    * @throws IllegalArgumentException if invalid parameter such
+    * as negative size or null file reference is passed.
     * @see JFileChooser#getIcon
     * @see AbstractMultiResolutionImage
     * @see FileSystemView#getSystemIcon(File)
     * @since 17
     */
-    public Icon getSystemIcon(File f, int size) {
-        if (f == null) {
+    public Icon getSystemIcon(File f, int width, int height) {
+        if (height < 1 || width < 1) {
+            throw new IllegalArgumentException("Icon size can not be below 1");
+        }
+
+        if (f == null){
+            throw new IllegalArgumentException("File reference should be specified");
+        }
+
+        if(!f.exists()) {
             return null;
```

Comment:
Shall it throw `FileNotFoundException` or `IllegalArgumentException` if the file doesn't exist?
It could more convenient to return `null` rather than catch an exception.

The space is missing between if and the opening parenthesis.

Reply:
It definitely should not be IAE. But FNFE is a reasonable idea.
However it changes the usage since it is a checked exception.
I'm on the fence and could go either way.

Reply:
The older similar method, `getSystemIcon(File f)`, returns `null` if the file doesn't exist. It could make sense to preserve the behaviour from this point of view and not to make the user of the API handle FNFE; thus the new method could easily be used instead.